### PR TITLE
Initial changes for e2e testing framework

### PR DESCRIPTION
**Pull request by patilpankaj212**
Added tests for init command.

### CODE REVIEW

**Comment by dev-gaur** [comment]
One test case I can think of :

1. execute terrascan init. Check if the timestamp of the downloaded $HOME/.terrascan directory. Let's say it was created at time t1.
2. execute terrascan init again. Check if the timestamp of the downloaded $HOME/.terrascan directory. Let's say it was created at time t2.

3. Assert that t2 > t1. This verifies that $HOME/.terrascan is downloaded fresh everytime `terrascan init` is run.

File: test/e2e/init/init_test.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+		When("the config file has invalid rego subdir", func() {
+			JustBeforeEach(func() {
+				os.Setenv(terrascanConfigEnvName, "config/invalid_rego_subdir.toml")
+			})
+			JustAfterEach(func() {
+				os.Setenv(terrascanConfigEnvName, "")
+			})
+			It("should error out and exit with status code 1", func() {
+				session = runInitCommand(terrascanBinaryPath, outWriter, errWriter, 1)
+				helper.ContainsErrorSubString(session, "invalid/path: no such file or directory")
```
Comment (dev-gaur) [inline test/e2e/init/init_test.go:149]: we should make changes to return a more specific error message for erranous rego subdir.
Not to be fixed in this PR though.

File: test/e2e/init/init_test.go [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+						Expect(err).ToNot(HaveOccurred())
+						Expect(fi.ModTime()).To(BeTemporally(">", modifiedTime))
+					})
+				})
+			})
+		})
+	})
+})
+
+func runInitCommand(terrascanBinaryPath string, outWriter, errWriter io.Writer, exitCode int) *gexec.Session {
```
Comment (dev-gaur) [inline test/e2e/init/init_test.go:207]: these three functions can be moved to helper or test_util file

**Review by dev-gaur — DISMISSED** [review]
Just a minor change to move the functions. everything else LGTM.

**Review by kanchwala-yusuf — APPROVED** [review]

**Pull request merged**